src/core/ViewCubeGizmo.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react'
+import { CanvasTexture, Event, Vector3 } from 'three'
+import { useGizmoHelper } from './GizmoHelper'
+
+const faces = ['RIGHT', 'LEFT', 'TOP', 'BOTTOM', 'FRONT', 'BACK']
+
+const edges: Array<[number, number, number]> = [
+  [1, 1, 0],
+  [1, 0, 1],
+  [1, 0, -1],
+  [1, -1, 0],
+  [0, 1, 1],
+  [0, 1, -1],
+  [0, -1, 1],
+  [0, -1, -1],
+  [-1, 1, 0],
+  [-1, 0, 1],
+  [-1, 0, -1],
+  [-1, -1, 0],
+]
+
+const corners: Array<[number, number, number]> = [
+  [1, 1, 1],
+  [1, 1, -1],
+  [1, -1, 1],
+  [1, -1, -1],
+  [-1, 1, 1],
+  [-1, 1, -1],
+  [-1, -1, 1],
+  [-1, -1, -1],
+]
+
+const FaceMaterial = ({ hover, index }: { hover: boolean; index: number }) => {
+  const texture = React.useMemo(() => {
+    const canvas = document.createElement('canvas')
+    canvas.width = 128
+    canvas.height = 128
+
+    const context = canvas.getContext('2d')!
+    context.fillStyle = '#eee'
+    context.fillRect(0, 0, canvas.width, canvas.height)
+    context.strokeStyle = '#444'
+    context.strokeRect(0, 0, canvas.width, canvas.height)
+
+    context.font = '28px Arial'
+    context.textAlign = 'center'
+    context.fillStyle = '#222'
+    context.fillText(faces[index], 64, 76)
+    return new CanvasTexture(canvas)
+  }, [index])
+  return <meshLambertMaterial map={texture} attachArray="material" color={hover ? 'hotpink' : 'white'} />
+}
+
+const FaceCube = () => {
+  const { tweenCamera, raycast } = useGizmoHelper()
+  const [hover, set] = React.useState<number | null>(null)
+  return (
+    <mesh
+      raycast={raycast}
+      onPointerOut={() => set(null)}
+      onPointerMove={(e: Event) => set(Math.floor(e.faceIndex / 2))}
+      onPointerDown={(e: Event) => void (tweenCamera(e.face.normal), e.stopPropagation())}
```

Comment:
imo inline handlers are hard to read, can you remove them out and give them corresponding names please? e.g `handlePointerMove`